ts/components/SelectCalendarModal.tsx

```diff
@@ -0,0 +1,108 @@
+import * as pot from "italia-ts-commons/lib/pot";
+import { Button, Container, Content, Text } from "native-base";
+import React from "react";
+import { FlatList, ListRenderItem, Modal } from "react-native";
+import RNCalendarEvents, { Calendar } from "react-native-calendar-events";
+import I18n from "../i18n";
+
+type CalendarItemProps = {
+  calendar: Calendar;
+  onPress: (calendar: Calendar) => void;
+};
+
+/**
+ * A component to render a Calendar as FlatList item
+ */
+class CalendarItem extends React.PureComponent<CalendarItemProps> {
+  public render() {
+    return <Text onPress={this.onPress}>{this.props.calendar.title}</Text>;
+  }
+
+  private onPress = () => {
+    const { calendar, onPress } = this.props;
+
+    onPress(calendar);
+  };
+}
+
+type Props = {
+  onCancel: () => void;
+  onCalendarSelected: (calendar: Calendar) => void;
+  header?: React.ReactNode;
+};
+
+type State = {
+  calendars: pot.Pot<ReadonlyArray<Calendar>, Error>;
```

Comment:
for the error, use a `string` type OR (better) an ADT and map to a string in the `render` method

Reply:
Done